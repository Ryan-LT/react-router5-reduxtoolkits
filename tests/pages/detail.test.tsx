import React from 'react';
import { Detail } from '@pages';
import { fireEvent } from '@testing-library/dom';
import { cleanup, render, screen } from '../render-test';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<Detail />);

  expect(asFragment()).toMatchSnapshot();
});

describe('<Detail/>', () => {
  it('render Detail component', () => {
    const component = render(<Detail />);
    expect(component).toBeTruthy();
  });

  it('should be default 0', () => {
    const { getByTestId } = render(<Detail />);
    expect(getByTestId('counter-value')).toBeEnabled();
    expect(getByTestId('counter-value')).toHaveTextContent('0');
  });

  it('increase counter', () => {
    const { getByTestId } = render(<Detail />);
    fireEvent.click(getByTestId('increase-button'));
    expect(getByTestId('counter-value')).toHaveTextContent('1');
  });

  it('decrease counter', () => {
    const { getByTestId } = render(<Detail />);
    fireEvent.click(getByTestId('decrease-button'));
    expect(getByTestId('counter-value')).toHaveTextContent('0');
  });

  it('decrease counter async', async () => {
    const component = render(<Detail />);
    const button = component.getByTestId('incease-async-button');
    fireEvent.click(button);
    fireEvent.click(button);
    fireEvent.click(button);
    fireEvent.click(button);
    const counter = await component.findByText('1');
    expect(counter).toHaveTextContent('4');
  });
});
