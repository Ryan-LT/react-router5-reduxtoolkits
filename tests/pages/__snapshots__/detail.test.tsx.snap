// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="increase-button"
      type="button"
    >
      Increase
    </button>
    <h1
      data-testid="counter-value"
    >
      0
    </h1>
    <button
      data-testid="decrease-button"
      type="button"
    >
      Decrease
    </button>
    <button
      type="button"
    >
      Navigate Back
    </button>
    <button
      data-testid="incease-async-button"
      type="button"
    >
      Async Increase
    </button>
  </div>
</DocumentFragment>
`;
